gcloud-java-compute/src/test/java/com/google/gcloud/compute/OperationTest.java

```diff
@@ -0,0 +1,489 @@
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(null);
+    replay(compute);
+    initializeOperation();
+    assertFalse(operation.exists());
+    verify(compute);
+  }
+
+  @Test
+  public void testIsDone_True() throws Exception {
+    initializeExpectedOperation(3);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    assertTrue(operation.isDone());
+    verify(compute);
+  }
+
+  @Test
+  public void testIsDone_False() throws Exception {
+    initializeExpectedOperation(4);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(
+        Operation.fromPb(serviceMockReturnsOptions, globalOperation.toPb().setStatus("PENDING")));
+    replay(compute);
+    initializeOperation();
+    assertFalse(operation.isDone());
+    verify(compute);
+  }
+  @Test
+  public void testIsDone_NotExists() throws Exception {
+    initializeExpectedOperation(3);
+    Compute.OperationOption[] expectedOptions =
+        {Compute.OperationOption.fields(Compute.OperationField.STATUS)};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, expectedOptions)).andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    assertTrue(operation.isDone());
+    verify(compute);
+  }
+
+  @Test
+  public void testReload() throws Exception {
+    initializeExpectedOperation(5);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID)).andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    Operation updatedOperation = operation.reload();
+    compareOperation(globalOperation, updatedOperation);
+    verify(compute);
+  }
+
+  @Test
+  public void testReloadNull() throws Exception {
+    initializeExpectedOperation(3);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID)).andReturn(null);
+    replay(compute);
+    initializeOperation();
+    assertNull(operation.reload());
+    verify(compute);
+  }
+
+  @Test
+  public void testReloadWithOptions() throws Exception {
+    initializeExpectedOperation(5);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.get(GLOBAL_OPERATION_ID, Compute.OperationOption.fields()))
+        .andReturn(globalOperation);
+    replay(compute);
+    initializeOperation();
+    Operation updatedOperation = operation.reload(Compute.OperationOption.fields());
+    compareOperation(globalOperation, updatedOperation);
```

Comment:
Similar to above, slightly weird to assert this considering the service would return a globalOperation with most fields set to null.